patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -1387,6 +1387,17 @@
  								SmartSelect_SelectItem(i);
  								return;
  						}
+@@ -12322,7 +_,9 @@
+ 		public static float GetClosestRollLuck(int x, int y, int range) => Main.player[FindClosest(new Vector2(x * 16, y * 16), 1, 1)].RollLuck(range);
+ 
+ 		public void ResetEffects() {
+-			if (extraAccessory && (Main.expertMode || Main.gameMenu))
++			if (extraAccessory && Main.masterMode)
++				extraAccessorySlots = 2;
```

Comment:
Please make sure this doesn't have unintended side effects. (Haven't checked myself)

Reply:
As far as I can tell, using Visual Studio's find references, the only part of the code that depends on extraAccessorySlots is in `Player.UpdateEquips` and `ItemLoader.GetWing`

The vanilla decompile also initially sets it to 2, before it gets set to 1 by `ResetEffects` so I don't think that will cause any issues.